src/Compiler/Microsoft.NET.Sdk.Razor.SourceGenerators/RazorSourceGenerator.cs

```diff
@@ -171,7 +185,7 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
                     var tagHelperFeature = new StaticCompilationTagHelperFeature();
                     var discoveryProjectEngine = GetDiscoveryProjectEngine(compilation.References.ToImmutableArray(), tagHelperFeature);
 
-                    List<TagHelperDescriptor> descriptors = new();
+                    var descriptors = ImmutableArray.CreateBuilder<TagHelperDescriptor>();
```

Comment:
I believe we should have access to `ArrayBuilder` now with Dustin's latest changes, so let's leverage that.

Reply:
Done.